src/libstore/http-binary-cache-store.cc

```diff
@@ -14,6 +14,14 @@ struct HttpBinaryCacheStoreConfig : virtual BinaryCacheStoreConfig
 
     const std::string name() override { return "HTTP Binary Cache Store"; }
 
+    static std::set<std::string> uriSchemes()
+    {
+        static bool forceHttp = getEnv("_NIX_FORCE_HTTP") == "1";
```

Comment:
if this is cached, just make it passed into the constructor prior to the config construction and not in the getter ?

Also, you'd think FORCE_HTTP only returns http.